### PR TITLE
Upgraded to Typescript@3.6, Vega@5.2, Vega-Lite@3.4, Webpack@4.1

### PR DESCRIPTION
**Pull request by siyaoL1**
List of upgraded dependencies:

- vega
- vega-lite
- compassql
- vega-datasets
- vega-expression
- vega-tooltip
- vega-typings
- vega-util
- vega-functions
- css-loader
- extract-text-webpack-plugin
- file-loader
- html-webpack-plugin
- postcss-loader
- react-dnd
- react-dev-utils
- react-hot-loader
- sass-loader
- source-map-loader
- terser-webpack-plugin
- ts-jest
- ts-loader
- typescript
- url-loader
- webpack
- webpack-cli
- webpack-dev-server
- webpack-manifest-plugin
- @types/webpack
- @types/node
- @types/react

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,34 +64,33 @@
     "redux": "^3.6.0",
     "redux-action-log": "^2.0.1",
     "redux-logger": "^3.0.6",
-    "redux-thunk": "^2.2.0",
+    "redux-thunk": "2.2.0",
```
Comment (domoritz) [inline package.json:67]: What's the problem with `^2.2.0`?

File: package.json [hunk 2]
```diff
@@ -64,34 +64,33 @@
...
-    "vega-lite": "2.4.0",
-    "vega-tooltip": "^0.4.2",
-    "vega-typings": "^0.2.15",
-    "vega-util": "1.7.0"
+    "vega": "5.2.0",
+    "vega-datasets": "^2.0.0",
+    "vega-expression": "^2.6.6",
+    "vega-lite": "3.4.0",
+    "vega-tooltip": "0.17.0",
+    "vega-typings": "0.7.2",
```
Comment (domoritz) [inline package.json:78]: Are those included in vega now? If yes, remove it here.

File: package.json [hunk 3]
```diff
@@ -140,7 +139,7 @@
   },
   "resolutions": {
     "@types/react": "^15.7.2",
-    "@types/node": "~8.0.22"
+    "@types/node": "~8.10.64"
```
Comment (domoritz) [inline package.json:142]: Can we move to a supported node version?

**Review by domoritz — CHANGES_REQUESTED** [review]
See comments

**Comment by siyaoL1** [comment]
I will work on addressing the comments

**Comment by domoritz** [comment]
Please fix the tests

**Comment by domoritz** [comment]
Let me know when tests pass.

File: config/webpack.config.lib.js [hunk 4]
```diff
@@ -6,10 +6,10 @@ const CaseSensitivePathsPlugin = require('case-sensitive-paths-webpack-plugin');
 const TerserPlugin = require("terser-webpack-plugin");
 var WebpackNotifierPlugin = require('webpack-notifier');
 
-const getClientEnvironment = require('./env');
+// const getClientEnvironment = require('./env');
```
Comment (domoritz) [inline config/webpack.config.lib.js:9]: please remove commented out code

File: config/webpack.config.lib.js [hunk 5]
```diff
@@ -6,10 +6,10 @@ const CaseSensitivePathsPlugin = require('case-sensitive-paths-webpack-plugin');
...
 // Get environment variables to inject into our app.
-var publicUrl = '';
-const env = getClientEnvironment(publicUrl);
+// var publicUrl = '';
+// const env = getClientEnvironment(publicUrl);
```
Comment (domoritz) [inline config/webpack.config.lib.js:12]: same as above

**Review by domoritz — APPROVED** [review]
Please remove [yarn-error.log](https://github.com/vega/voyager/pull/886/files#diff-a2b6efbfdcf1ef38207392bbe0d2316d07f49cb87184999c894e9535e2eb65cc). 

Did you check that the app works as expected? If so, I will make a final pass and merge after you address my comments.

**Comment by domoritz** [comment]
@siyaoL1 is working on support CSV and then we can merge this.

**Pull request closed without merging**